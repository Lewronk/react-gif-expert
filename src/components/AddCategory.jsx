import { useState } from 'react'
import { PropTypes } from "prop-types";

export const AddCategory = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ( {target} ) => {
        setInputValue( target.value )
    }


    //Evento onsubmit
    const onSubmit = (event) => {
      
      //Evitamos la recarga default del enter en form
      event.preventDefault();

      //Mas de un caracter o no hacemos nada
      if(inputValue.trim().length <= 1) return;

      //Desestructurado y agregado del item
      //setCategories( categories => [ inputValue, ...categories ])

      //Mandando al elemento padre
      onNewCategory( inputValue.trim() );

      //Limpiamos el input
      setInputValue('');

      //console.log(inputValue);
    }

    //(event) => onSubmit(event)

  return (

    <form onSubmit={ onSubmit }>
      <input
        type='text'
        placeholder='Buscar gifs'
        value={inputValue}
        onChange={ onInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}