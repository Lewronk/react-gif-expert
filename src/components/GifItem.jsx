import React from 'react'
import PropTypes from "prop-types";

export const GifItem = ({tittle, url}) => {

  return (
    //<li key={id}>{tittle}</li>
    <div className='card'>
        <img src={url} alt={tittle} />
        <p>{tittle}</p>
    </div>
  )
}

GifItem.PropTypes = {
  tittle: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}