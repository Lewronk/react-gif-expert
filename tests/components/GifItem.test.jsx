import { render, screen } from '@testing-library/react'
import { GifItem } from "../../src/components/GifItem"

describe('Prueba en GifItem', () => {

    const titulo = 'Titulo Prueba'
    const url = 'http://localhost/Url%20bien%20chingona'

    test('Debe hacer match con el Snapshot', () => {
        
        const { container } = render(<GifItem tittle={titulo} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el URL y el ALT indicado', () => {

        const { container } = render(<GifItem tittle={titulo} url={url} />);
        //screen.debug();
        //expect( screen.getByRole('img').src ).toBe( url );
        //expect( screen.getByRole('img').alt ).toBe( tittle );

        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(titulo);

    })

    test('debe mostrar el titulo en el componente', () => {

        render(<GifItem tittle={titulo} url={url} />);
        expect( screen.getByText(titulo) ).toBeTruthy();
    })
})
