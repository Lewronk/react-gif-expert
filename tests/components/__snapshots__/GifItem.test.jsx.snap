// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifItem Debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo Prueba"
      src="http://localhost/Url%20bien%20chingona"
    />
    <p>
      Titulo Prueba
    </p>
  </div>
</div>
`;
